osf_tests/factories.py

```diff
@@ -395,6 +395,7 @@ def add_approval_step(reg):
                 reg.sanction.save()
         if is_public:
             reg.is_public = True
+        reg.files_count = reg.registered_from.files.filter(deleted_on__isnull=True).count()
```

Comment:
Since we are mocking all a good part of the archiving process in tests,  `registered_from` project files are not transferred to the registration. Given that `files_count` is updated during (more like at the end of) the archiving process, this is the only way we could think of to simulate the actual.

Reply:
makes sense!